src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/service/IslRulesService.java

```diff
@@ -0,0 +1,145 @@
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.network.service;
+
+import static java.lang.String.format;
+
+import org.openkilda.floodlight.api.request.rulemanager.FlowCommand;
+import org.openkilda.floodlight.api.request.rulemanager.GroupCommand;
+import org.openkilda.floodlight.api.request.rulemanager.MeterCommand;
+import org.openkilda.floodlight.api.request.rulemanager.OfCommand;
+import org.openkilda.floodlight.api.response.rulemanager.SpeakerCommandResponse;
+import org.openkilda.model.SwitchId;
+import org.openkilda.persistence.PersistenceManager;
+import org.openkilda.rulemanager.DataAdapter;
+import org.openkilda.rulemanager.FlowSpeakerData;
+import org.openkilda.rulemanager.GroupSpeakerData;
+import org.openkilda.rulemanager.MeterSpeakerData;
+import org.openkilda.rulemanager.RuleManager;
+import org.openkilda.rulemanager.SpeakerData;
+import org.openkilda.rulemanager.adapter.PersistenceDataAdapter;
+import org.openkilda.wfm.share.model.Endpoint;
+import org.openkilda.wfm.share.model.IslReference;
+
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@Slf4j
+public class IslRulesService {
+    private final IIslCarrier carrier;
+    private final PersistenceManager persistenceManager;
+    private final RuleManager ruleManager;
+
+    private final Map<UUID, Request> requests = new HashMap<>();
+
+    public IslRulesService(IIslCarrier carrier, PersistenceManager persistenceManager, RuleManager ruleManager) {
+        this.carrier = carrier;
+        this.persistenceManager = persistenceManager;
+        this.ruleManager = ruleManager;
+    }
+
+    /**
+     * Build and send required ISL service rules install commands.
+     */
+    public void installIslRules(IslReference reference, Endpoint endpoint) {
+        List<OfCommand> commands = buildIslRules(endpoint.getDatapath(), endpoint.getPortNumber());
+        UUID commandId = UUID.randomUUID();
+        requests.put(commandId, new Request(reference, endpoint, Action.INSTALL));
+        carrier.sendIslRulesInstallCommand(endpoint.getDatapath(), commandId, commands);
+    }
+
+    /**
+     * Build and send required ISL service rules delete commands.
+     */
+    public void deleteIslRules(IslReference reference, Endpoint endpoint) {
+        List<OfCommand> commands = buildIslRules(endpoint.getDatapath(), endpoint.getPortNumber());
+        UUID commandId = UUID.randomUUID();
+        requests.put(commandId, new Request(reference, endpoint, Action.DELETE));
+        carrier.sendIslRulesDeleteCommand(endpoint.getDatapath(), commandId, commands);
+    }
+
+    /**
+     * Process ISL service rules installed/deleted response.
+     */
+    public void handleResponse(SpeakerCommandResponse response) {
```

Comment:
Much better... 

PS Can we use some lambda (or adapter) instead of `request.action`?

Reply:
Reworked to use lambda.